File: frontend/src/lib/types/styleguide.ts
// Style Guide types

export interface StyleGuide {
  id: string
  name: string
  description: string
  guide_type: string // report, landing_page, two_pager, memo

  // Typography
  title_font: string
  body_font: string
  title_size: string
  heading_size: string
  subheading_size: string
  body_size: string
  line_spacing: string

  // Branding
  logo_url: string
  strapline: string
  primary_color: string
  secondary_color: string
  accent_color: string

  // Layout
  page_size: string // letter, a4, custom
  page_orientation: string // portrait, landscape
  margin_top: string
  margin_bottom: string
  margin_left: string
  margin_right: string

  // Content Rules
  heading_style: string // bold, underline, caps
  color_scheme: string // dark, light, brand
  include_toc: boolean
  include_page_numbers: boolean

  created?: string
  updated?: string
}

export interface StyleGuideCreate {
  name: string
  description?: string
  guide_type?: string
  title_font?: string
  body_font?: string
  title_size?: string
  heading_size?: string
  subheading_size?: string
  body_size?: string
  line_spacing?: string
  logo_url?: string
  strapline?: string
  primary_color?: string
  secondary_color?: string
  accent_color?: string
  page_size?: string
  page_orientation?: string
  margin_top?: string
  margin_bottom?: string
  margin_left?: string
  margin_right?: string
  heading_style?: string
  color_scheme?: string
  include_toc?: boolean
  include_page_numbers?: boolean
}

export type StyleGuideUpdate = StyleGuideCreate

export const GUIDE_TYPES = [
  { value: 'report', label: 'Report (8.5×11)', description: 'Full-page document with TOC' },
  { value: 'two_pager', label: 'Two-Pager', description: 'Executive summary brief' },
  { value: 'landing_page', label: 'Landing Page', description: 'Single-page web layout' },
  { value: 'memo', label: 'Memo', description: 'Internal communication format' },
] as const

export const PAGE_SIZES = [
  { value: 'letter', label: 'US Letter (8.5×11")' },
  { value: 'a4', label: 'A4 (210×297mm)' },
  { value: 'custom', label: 'Custom' },
] as const

export const HEADING_STYLES = [
  { value: 'bold', label: 'Bold' },
  { value: 'underline', label: 'Underlined' },
  { value: 'caps', label: 'ALL CAPS' },
] as const

export const COLOR_SCHEMES = [
  { value: 'dark', label: 'Dark Mode' },
  { value: 'light', label: 'Light Mode' },
  { value: 'brand', label: 'Brand Colors' },
] as const

File: open_notebook/domain/styleguide.py
from typing import ClassVar, Optional

from open_notebook.domain.base import ObjectModel


class StyleGuide(ObjectModel):
    table_name: ClassVar[str] = "styleguide"
    nullable_fields: ClassVar[set[str]] = {
        "description",
        "guide_type",
        "title_font",
        "body_font",
        "title_size",
        "heading_size",
        "subheading_size",
        "body_size",
        "line_spacing",
        "logo_url",
        "strapline",
        "primary_color",
        "secondary_color",
        "accent_color",
        "page_size",
        "page_orientation",
        "margin_top",
        "margin_bottom",
        "margin_left",
        "margin_right",
        "heading_style",
        "color_scheme",
    }

    name: str = ""
    description: Optional[str] = ""
    guide_type: Optional[str] = "report"  # report, landing_page, two_pager, memo

    # Typography
    title_font: Optional[str] = "Inter"
    body_font: Optional[str] = "Inter"
    title_size: Optional[str] = "24pt"
    heading_size: Optional[str] = "18pt"
    subheading_size: Optional[str] = "14pt"
    body_size: Optional[str] = "11pt"
    line_spacing: Optional[str] = "1.5"

    # Branding
    logo_url: Optional[str] = ""
    strapline: Optional[str] = ""
    primary_color: Optional[str] = "#1a73e8"
    secondary_color: Optional[str] = "#34a853"
    accent_color: Optional[str] = "#fbbc04"

    # Layout
    page_size: Optional[str] = "letter"  # letter (8.5x11), a4, custom
    page_orientation: Optional[str] = "portrait"
    margin_top: Optional[str] = "1in"
    margin_bottom: Optional[str] = "1in"
    margin_left: Optional[str] = "1in"
    margin_right: Optional[str] = "1in"

    # Content Rules
    heading_style: Optional[str] = "bold"  # bold, underline, caps
    color_scheme: Optional[str] = "dark"  # dark, light, brand
    include_toc: Optional[bool] = True
    include_page_numbers: Optional[bool] = True
